x-pack/plugins/infra/public/pages/metrics/metric_detail/components/side_nav.tsx

```diff
@@ -32,22 +31,23 @@ export const MetricsSideNav = ({ loading, name, items }: Props) => {
       isOpenOnMobile={isOpenOnMobile}
     />
   );
+
   return (
-    <EuiPageSideBar>
-      <EuiHideFor sizes={['xs', 's']}>
-        <SideNavContainer>{content}</SideNavContainer>
+    <>
+      <EuiHideFor sizes={['xs', 's', 'm']}>
+        <StickyPageSideBar>{content}</StickyPageSideBar>
       </EuiHideFor>
-      <EuiShowFor sizes={['xs', 's']}>{mobileContent}</EuiShowFor>
-    </EuiPageSideBar>
+      <EuiShowFor sizes={['xs', 's', 'm']}>
+        <EuiPageSideBar>{mobileContent}</EuiPageSideBar>
+      </EuiShowFor>
+    </>
   );
 };
 
-const SideNavContainer = euiStyled.div`
-  position: fixed;
-  z-index: 1;
-  height: 88vh;
-  padding-left: 16px;
-  margin-left: -16px;
-  overflow-y: auto;
-  overflow-x: hidden;
+const StickyPageSideBar = euiStyled(EuiPageSideBar)`
+  position: sticky;
+  top: ${(props) =>
+    parseFloat(props.theme.eui.euiHeaderHeightCompensation) * 2 +
```

Comment:
I will try and investigate this a bit more. I don't like magic numbers but was struggling to find the "origin" for this one. There's lots of places doing this * 2 in Kibana to account for the header fixed space, but I struggled to find why there wasn't a separate variable for that.

Reply:
It's because there are **two** headers 😄 . But more importantly, why are these extra styles necessary? Is the template not behaving correctly? This should all be accounted for.

Reply:
@cchaos Oh, duh 😅, thanks! 

This is a use case on the node details page in Metrics. There's a loose design here: https://github.com/elastic/kibana/issues/99970.

The additional side navigation with the metrics uses a fixed position. 

![fixed-nav](https://user-images.githubusercontent.com/471693/121216221-9b2d4c80-c878-11eb-8b55-dfe43f5d35f6.gif)

Would there be a better way to do this? Or do you recommend removing the fixed nature?

Reply:
OH it's a secondary side-nav, not the main Solution nav. I see. Does applying `position="sticky"` to the EuiPageSideBar component not work? That should then ensure that the top position is always correct because of [this mixin](https://github.com/elastic/kibana/blob/356b7d6c9202269d00b70886f95bfe0eb97555c5/src/core/public/rendering/_base.scss#L40-L59).

Reply:
Ah - I somehow missed the `sticky` prop on the `<EuiPageSideBar />`. That works 🙌 

Thanks for the input.